### PR TITLE
improve(RedisUtils): Add delay to cache historical deposits

### PR DESCRIPTION
**Pull request by nicholaspai**
Mitigates chance of caching deposit that gets re-orged

### CODE REVIEW

File: src/clients/SpokePoolClient.ts [hunk 1]
```diff
@@ -343,7 +343,14 @@ export class SpokePoolClient {
         deposit,
       });
       if (this.configStoreClient.redisClient)
-        await setDeposit(deposit, this.configStoreClient.redisClient, 24 * 60 * 60);
+        await setDeposit(
+          deposit,
+          (
+            await this.spokePool.provider.getBlock("latest")
```
Comment (mrice32) [inline src/clients/SpokePoolClient.ts:349]: Why not use the current system timestamp?

**Review by mrice32 — COMMENTED** [review]
Looks good! Just one question

**Comment by nicholaspai** [comment]
@mrice32 do you think we even need to cache this deposit now that we have the provider cache?

File: src/clients/SpokePoolClient.ts (hunk 1 above)
Comment (nicholaspai) [reply src/clients/SpokePoolClient.ts]: done

**Comment by mrice32** [comment]
> @mrice32 do you think we even need to cache this deposit now that we have the provider cache? If not I can open a new PR to remove deposit caching and reduce complexity

I think we do, mainly because when we binary search blocks, we always take a different path, so I suspect that many of the blocks will not be cached on each run. So caching the result avoids the search entirely. We also don't cache eth_getBlockByNumber or eth_call right now, so removing it today would result in us not caching this at all.

However, this isn't a super strong opinion, I just suspect that this caching is adding value.

Thoughts?

**Review by mrice32 — APPROVED** [review]

**Pull request merged**